controllers/etcd_controller.go

```diff
@@ -1150,16 +1129,18 @@ func (r *EtcdReconciler) updateEtcdErrorStatus(ctx context.Context, etcd *druidv
 		etcd.Status.Ready = &ready
 	}
 
-	if err := r.Status().Update(ctx, etcd); err != nil && !apierrors.IsNotFound(err) {
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		return r.Status().Update(ctx, etcd)
```

Comment:
This will unfortunately not work if there is a conflict because you need to re-fetch the current `etcd` resource from the cluster (to get up-to-date `ResourceVersion`), transform it and call `Update` again.

The Gardener utility functions [TryUpdate](https://github.com/gardener/gardener/blob/07a56785cb69aff4a1529943dfc8f5d6ec97f015/pkg/utils/kubernetes/update.go#L29) and [TryUpdateStatus](https://github.com/gardener/gardener/blob/07a56785cb69aff4a1529943dfc8f5d6ec97f015/pkg/utils/kubernetes/update.go#L35) handle this automatically for you (also `DeepCopy()` is not required before). You can see an example usage [here](https://github.com/gardener/gardener/blob/77e09dd522d7bfb682cc35d0be437201b606061b/extensions/pkg/controller/worker/state_actuator.go#L71).

The same applies for other status updates in this PR.

Reply:
I did. Thanks.